Bottom: c722f7304a7b4d6066f096999ce0cc9128453e8b
Top:    34aea2338028a7b95d4f4ed946e0896de4655319
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-07 22:41:03 +0100

action.c: Simplify ActionMovePointer()

The save / restore of the crosshair coordinates is unnecessary since
we are emitting the appropriate change notifications.


---

diff --git a/src/action.c b/src/action.c
index e222dbf..6339e39 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2114,18 +2114,11 @@ ActionSetThermal (int argc, char **argv, int x, int y)
 void
 ActionMovePointer (char *deltax, char *deltay)
 {
-  LocationType x, y, dx, dy;
+  LocationType dx, dy;
 
-  /* save old crosshair position */
-  x = Crosshair.X;
-  y = Crosshair.Y;
   dx = (LocationType) (atoi (deltax) * PCB->Grid);
   dy = (LocationType) (atoi (deltay) * PCB->Grid);
-  MoveCrosshairRelative (TO_SCREEN_SIGN_X (dx), TO_SCREEN_SIGN_Y (dy));
-  FitCrosshairIntoGrid (Crosshair.X, Crosshair.Y);
-  /* restore crosshair for erasure */
-  Crosshair.X = x;
-  Crosshair.Y = y;
+
   notify_crosshair_change (false);
   MoveCrosshairRelative (TO_SCREEN_SIGN_X (dx), TO_SCREEN_SIGN_Y (dy));
   /* update object position and cursor location */
